### PR TITLE
DOCSP-42956: Remove $collection support

### PR DESCRIPTION
**Pull request by norareidy**
JIRA - https://jira.mongodb.org/browse/DOCSP-42956
Staging - https://deploy-preview-136--docs-laravel.netlify.app/upgrade/#version-5.x-breaking-changes

### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file

### CODE REVIEW

File: docs/upgrade.txt [hunk 1]
```diff
@@ -79,6 +79,14 @@ This library version introduces the following breaking changes:
   of this behavior, you cannot have two separate ``id`` and ``_id`` fields in your
   documents.
 
+- Removes support for the ``$collection`` property and the following associated
+  methods:
+  
+  - ``Model::collection()``. Instead, use ``Model::table()``.
+  - ``DB::collection()``. Instead, use ``DB::table()``.
+  - ``Connection::collection()``. Instead, use ``Connection::table()``.
+  - ``Schema::collection()``. Instead, use ``Schema::table()``.
```
Comment (rustagir) [inline docs/upgrade.txt]: S: this list with periods after the method names seems awkward. what about something like
```suggestion
- Removes support for the ``$collection`` property and associated
  methods. The following list contains the removed methods and provides alternative methods
  that you can use to replace the functionality:
  
  - ``Model::collection()``: use ``Model::table()``
  - ``DB::collection()``: use ``DB::table()``
  - ``Connection::collection()``: use ``Connection::table()``
  - ``Schema::collection()``: use ``Schema::table()``
```

**Review by rustagir — APPROVED** [review]
lgtm with a wording change suggestion

File: docs/upgrade.txt [hunk 2]
```diff
@@ -79,6 +79,15 @@ This library version introduces the following breaking changes:
...
   documents.
 
+- Removes support for the ``$collection`` property and associated methods.
+  The following list contains the removed methods and provides alternative methods
+  that you can use to replace the functionality:
+  
+  - ``Model::collection()``: use ``Model::table()``
+  - ``DB::collection()``: use ``DB::table()``
+  - ``Connection::collection()``: use ``Connection::table()``
+  - ``Schema::collection()``: use ``Schema::table()``
```
Comment (GromNaN) [inline docs/upgrade.txt]: Actually there is only 2 methods removed in `MongoDB\Laravel\Connection` and `MongoDB\Laravel\Schema\Builder`, where they can be used as instance method.

But they are exposed in the Laravel facades `Illuminate\Support\Facades\DB` and `Illuminate\Support\Facades`, which expose them as static method. (Yes, this is very confusing).

```suggestion
  - ``DB::collection()``: use ``DB::table()``
  - ``Schema::collection()``: use ``Schema::table()``
```

**Review by GromNaN — CHANGES_REQUESTED** [review]

File: docs/upgrade.txt (hunk 2 above)
Comment (norareidy) [reply docs/upgrade.txt]: Okay I see, updated! Should I mention the static methods here?

**Review by GromNaN — COMMENTED** [review]
I think it would be better with a diff of the changes to be made.

```diff
use MongoDB\Laravel\Eloquent\Model;

class User extends Model
{
    protected $keyType = 'string';
-   protected $collection = 'app_user';
+   protected $table = 'app_user';

    ...
}
```


```diff
use Illuminate\Support\Facades\DB;

- DB::collection('app_user')->find($id);
+ DB::table('app_user')->find($id);
```

```diff
use Illuminate\Support\Facades\Schema;
use MongoDB\Laravel\Schema\Blueprint;

- Schema::collection('app_user', function (Blueprint $collection) { ... });
+ Schema::table('app_user', function (Blueprint $table) { ... });
```

**Review by GromNaN — APPROVED** [review]

**Pull request merged**